contracts/StablecoinConverter.sol

```diff
@@ -0,0 +1,119 @@
+pragma solidity ^0.5.0;
+
+import "./EpochTokenLocker.sol";
+import "./libraries/IdToAddressBiMap.sol";
+import "solidity-bytes-utils/contracts/BytesLib.sol";
+import "./libraries/IdToAddressBiMap.sol";
+
+
+contract StablecoinConverter is EpochTokenLocker {
+    using SafeMath for uint;
+    using BytesLib for bytes32;
+
+    event OrderPlacement(
+        address owner,
+        uint16 buyToken,
+        uint16 sellToken,
+        bool sellOrderFlag,
+        uint32 validFrom,
+        uint32 validUntil,
+        uint256 buyAmount,
+        uint256 sellAmount
+    );
+
+    event OrderCancelation(
+        address owner,
+        uint id
+    );
+
+    //outstanding volume of an order is encoded as buyAmount or sellAmount depending on sellOrderFlag
+    struct Order {
+        uint16 buyToken;
+        uint16 sellToken;
+        uint32 validFrom;  // order is valid from validFrom inclusively
+        uint32 validUntil;  // order is valid till validUntil inclusively
+        bool sellOrderFlag;
+        uint128 buyAmount;
+        uint128 sellAmount;
+    }
+
+    // User-> Order
+    mapping(address => Order[]) public orders;
```

Comment:
It would be good to have an unique id per order, this way you can identify it just with that and it's easier to refer to one order. 

This mappings needs an `address + index` to find an order

Reply:
But then, how would you return orders on-chain?
 Like how do you write a solidity function giving you all the orders for a user back?

We could use:
`    mapping( sha(address,index) => Order) public orders;
`
but then you would never know whether you reached the end of the order list.

Reply:
Would it be super bad to keep an array of user, where the position is the id of the order. So we have an unique id for every order (useful for matching, identification, etc..)

Then, for easily listing the orders of an user, would be to keep an `mapping(address => uint[])`

Definitely my proposal has some replication, not 100% sure if it's worth it, but it's simple enough and it make querying easier, and also make it easy to get on chain the most important entity we have, orders :)

Reply:
We want to introduce a hashmap for users so that later on we can query all orders also in a straight forward manner. 

In a sense we also have in the current implementation a unique identifier made up from( address+ id).

I prefer an identifier made up of two components instead of having replication.